src/Compilers/VisualBasic/Portable/Symbols/Source/SourceParameterSymbol.vb

```diff
@@ -264,6 +264,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return MyBase.EarlyDecodeWellKnownAttribute(arguments)
         End Function
 
+        Friend Overrides Iterator Function GetCustomAttributesToEmit(compilationState As ModuleCompilationState) As IEnumerable(Of VisualBasicAttributeData)
+            Dim attributes = MyBase.GetCustomAttributesToEmit(compilationState)
+            For Each attribute In attributes
+                If AttributeData.IsTargetEarlyAttribute(attributeType:=attribute.AttributeClass, attributeArgCount:=attribute.CommonConstructorArguments.Length, description:=AttributeDescription.CallerArgumentExpressionAttribute) Then
+                    Dim callerArgumentExpressionParameterIndex = Me.CallerArgumentExpressionParameterIndex
+                    If callerArgumentExpressionParameterIndex <> -1 AndAlso TypeOf attribute Is SourceAttributeData Then
+                        Debug.Assert(callerArgumentExpressionParameterIndex >= 0)
+                        Debug.Assert(attribute.CommonConstructorArguments.Length = 1)
+                        ' We allow CallerArgumentExpression to have case-insensitive parameter name, but we
+                        ' want to emit the parameter name with correct casing, so that it works with C#.
+                        Dim correctedParameterName = ContainingSymbol.GetParameters()(callerArgumentExpressionParameterIndex).Name
```

Comment:
Consider just returning the original attribute if no corrections need to be made.